### ISSUE TITLE
Cannot connect wtih xt/*.pl

### ISSUE THREAD

**Issue by kthakore** [tsee/ZeroMQ-Perl#4]
perl xt/local_lat.pl inproc://#1 100 2

perl xt/remote_lat.pl inproc://#1 100 2
Caught C++ exception of type or derived from 'zmq::error_t': Connection refused at xt/remote_lat.pl line 18.

**Comment by kthakore** [comment]
I was trying this example out http://api.zeromq.org/zmq_inproc.html

**Comment by tsee** [comment]
You're trying to do in-process communication in multiple processes. Try a TCP socket.

**Comment by kthakore** [comment]
Will it work if I do threads?

**Comment by tsee** [comment]
That's still untested as are ithreads in general. I would happily accept a test file!

**Comment by kthakore** [comment]
Um ... I can't use your extutils-xspp lengthfix (doesn't build and is 0.11_01) your branch for zeromq-perl is asking for 0.11_02. So I am going to make this ithreads test files in my branch. Also why are you not just binding to XS (0mq is a C library)? It would be a lot easier and stable code.

**Comment by tsee** [comment]
I just changed zeromq-perl to ask for 0.11_02 which will be the next release tentatively including the lengthFix. Furthermore, I hacked up an XS++ plugin that allows for controlling the behaviour while spawning a thread. With that in place, making an XS module thread safe is just putting "%loadplugin{Cloning}" at the top of the XS++ and marking a class as non-cloning. For me, it's not easier to work with C. The C++ interface is much more natural. I also prefer XS++ to XS (particularly for C++) since OO in XS is a horror. If you do not like the choice, you are free to use any of the code in a fork (under a different namespace) that uses plain XS and C.
